projects/admin/src/app/api/ill-request-api.service.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2023 RERO
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { Injectable } from '@angular/core';
+import { Record, RecordService } from '@rero/ng-core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { ILLRequestStatus } from '../classes/ill-request';
+import { BaseApi } from '@rero/shared';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class IllRequestApiService extends BaseApi {
+
+  /** Resource name */
+  readonly RESOURCE_NAME = 'ill_requests';
+
+  /**
+   * Constructor
+   * @param _recordService - RecordService
+   */
+  constructor(private _recordService: RecordService) {
+    super();
+  }
+
+  getByPatronPid(patronPid: string): Observable<Record | Error> {
```

Comment:
Return statement seems incorrect. It should return `Observable<Array<Record> | Error>`

Reply:
The signature of the `getRecords` function is `Observable<Record | Error>`.
Ref: https://github.com/rero/ng-core/blob/staging/projects/rero/ng-core/src/lib/record/record.service.ts#L116